tests/contrib/urllib3/test_urllib3.py

```diff
@@ -0,0 +1,386 @@
+        assert out.status == 200
+        spans = self.pop_spans()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.get_tag(http.METHOD) == "GET"
+        assert s.get_tag(http.URL) == URL_200
+        assert s.get_tag(http.STATUS_CODE) == "200"
+        assert s.error == 0
+        assert s.span_type == "http"
+        assert http.QUERY_STRING not in s.meta
+
+    def test_200_query_string(self):
+        """Tests query string tag is added when trace_query_string config is set"""
+        query_string = "key=value&key2=value2"
+        URL_200_QS = URL_200 + "?" + query_string
+        with self.override_http_config("urllib3", dict(trace_query_string=True)):
+            out = self.http.request("GET", URL_200_QS)
+            assert out.status == 200
+
+        spans = self.pop_spans()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.get_tag(http.METHOD) == "GET"
+        assert s.get_tag(http.STATUS_CODE) == "200"
+        assert s.get_tag(http.URL) == URL_200_QS
+        assert s.error == 0
+        assert s.span_type == "http"
+        assert s.get_tag(http.QUERY_STRING) == query_string
+
+    def test_post_500(self):
+        """Test a request with method POST and expected status 500"""
+        out = self.http.request("POST", URL_500)
+        assert out.status == 500
+        spans = self.pop_spans()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.get_tag(http.METHOD) == "POST"
+        assert s.get_tag(http.STATUS_CODE) == "500"
+        assert s.get_tag(http.URL) == URL_500
+        assert s.error == 1
+
+    def test_connection_retries(self):
+        """Tests a connection error results in error spans with proper exc info"""
+        retries = 3
+        try:
+            self.http.request("GET", "http://fakesubdomain." + SOCKET, retries=retries)
+        except Exception:
+            pass
+        else:
+            assert 0, "expected error"
+
+        spans = self.pop_spans()
+        assert len(spans) == 4  # Default retry behavior is 3 retries + original request
+        for i, s in enumerate(spans):
+            assert s.get_tag(http.METHOD) == "GET"
+            if i > 0:
+                assert s.get_tag(http.RETRIES_REMAIN) == str(retries - i)
+            assert s.error == 1
+            assert "Failed to establish a new connection" in s.get_tag(errors.MSG)
+            assert "Failed to establish a new connection" in s.get_tag(errors.STACK)
+            assert "Traceback (most recent call last)" in s.get_tag(errors.STACK)
+            assert "urllib3.exceptions.MaxRetryError" in s.get_tag(errors.TYPE)
+
+    def test_default_service_name(self):
+        """Test the default service name is set"""
+        out = self.http.request("GET", URL_200)
+        assert out.status == 200
+        spans = self.pop_spans()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.service == config.urllib3["_default_service"]
```

Comment:
```suggestion
        assert s.service == "urllib3"
```

Having the literal values helps make sure that we don't accidentally change values that users may depend on